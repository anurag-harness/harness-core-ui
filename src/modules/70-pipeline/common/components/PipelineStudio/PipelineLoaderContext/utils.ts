import { DefaultNewPipelineId, RouteState } from '@pipeline/components/PipelineStudio/PipelineContext/PipelineActions'
import { EntityGitDetails } from 'services/cd-ng'
import { getId, getRepoIdentifierName } from '@pipeline/components/PipelineStudio/PipelineContext/utils'
import { IDB } from '@modules/10-common/components/IDBContext/IDBContext'

export const deletePipelineFromIDB = async <T = unknown>({
  routeState,
  identifier,
  gitDetails,
  idb
}: {
  routeState: RouteState
  identifier: string
  gitDetails?: EntityGitDetails
  idb: IDB<T>
}): Promise<void> => {
  const id = getId(
    routeState.accountIdentifier,
    routeState.orgIdentifier || '',
    routeState.projectIdentifier || '',
    identifier,
    getRepoIdentifierName(gitDetails),
    gitDetails?.branch || ''
  )
  await idb.del(id)

  const defaultId = getId(
    routeState.accountIdentifier,
    routeState.orgIdentifier || '',
    routeState.projectIdentifier || '',
    DefaultNewPipelineId,
    getRepoIdentifierName(gitDetails),
    gitDetails?.branch || ''
  )
  await idb.del(defaultId)
}
