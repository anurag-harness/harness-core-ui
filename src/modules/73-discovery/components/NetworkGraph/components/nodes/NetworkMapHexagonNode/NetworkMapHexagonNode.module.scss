/*
 * Copyright 2023 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

.nodeContainer {
  width: 90px;
  height: 97px;
}

.hexagon {
  display: flex;
  align-items: center;
  justify-content: center;
  width: 100%;
  height: 100%;
  background-color: var(--blue-100);
  font-size: 13px;
  text-align: center;
  padding: 18px;
  cursor: pointer;

  clip-path: polygon(
    calc(50% - 1.25px) calc(50% - 47.20096px),
    calc(50% - 1.25px) calc(50% - 47.20096px),
    calc(50% - 1.012924px) calc(50% - 47.3215366px),
    calc(50% - 0.76723199999999px) calc(50% - 47.4153184px),
    calc(50% - 0.515078px) calc(50% - 47.4823054px),
    calc(50% - 0.258616px) calc(50% - 47.5224976px),
    calc(50% - 0px) calc(50% - 47.535895px),
    calc(50% - -0.258616px) calc(50% - 47.5224976px),
    calc(50% - -0.515078px) calc(50% - 47.4823054px),
    calc(50% - -0.76723200000001px) calc(50% - 47.4153184px),
    calc(50% - -1.012924px) calc(50% - 47.3215366px),
    calc(50% - -1.25px) calc(50% - 47.20096px),
    84.8192px 25.567px,
    84.8192px 25.567px,
    85.0421605px 25.7120301px,
    85.246224px 25.8779168px,
    85.4303135px 26.0627947px,
    85.593352px 26.2647984px,
    85.7342625px 26.4820625px,
    85.851968px 26.7127216px,
    85.9453915px 26.9549103px,
    86.013456px 27.2067632px,
    86.0550845px 27.4664149px,
    86.0692px 27.732px,
    86.0692px 72.2679px,
    86.0692px 72.2679px,
    86.0550845px 72.5335122px,
    86.013456px 72.7931856px,
    85.9453915px 73.0450554px,
    85.851968px 73.2872568px,
    85.7342625px 73.517925px,
    85.593352px 73.7351952px,
    85.4303135px 73.9372026px,
    85.246224px 74.1220824px,
    85.0421605px 74.2879698px,
    84.8192px 74.433px,
    46.25px 96.701px,
    46.25px 96.701px,
    46.012924px 96.821555px,
    45.767232px 96.91532px,
    45.515078px 96.982295px,
    45.258616px 97.02248px,
    45px 97.035875px,
    44.741384px 97.02248px,
    44.484922px 96.982295px,
    44.232768px 96.91532px,
    43.987076px 96.821555px,
    43.75px 96.701px,
    5.18078px 74.433px,
    5.18078px 74.433px,
    4.9578195px 74.2879699px,
    4.753756px 74.1220832px,
    4.5696665px 73.9372053px,
    4.406628px 73.7352016px,
    4.2657175px 73.5179375px,
    4.148012px 73.2872784px,
    4.0545885px 73.0450897px,
    3.986524px 72.7932368px,
    3.9448955px 72.5335851px,
    3.93078px 72.268px,
    3.93078px 27.7321px,
    3.93078px 27.7321px,
    3.9448955px 27.4664878px,
    3.986524px 27.2068144px,
    4.0545885px 26.9549446px,
    4.148012px 26.7127432px,
    4.2657175px 26.482075px,
    4.406628px 26.2648048px,
    4.5696665px 26.0627974px,
    4.753756px 25.8779176px,
    4.9578195px 25.7120302px,
    5.18078px 25.567px,
    calc(50% - 1.25px) calc(50% - 47.20096px)
  );
}

.targetHandle {
  background-color: var(--red-400) !important;
}

.sourceHandle {
  background-color: var(--primary-6) !important;
}
