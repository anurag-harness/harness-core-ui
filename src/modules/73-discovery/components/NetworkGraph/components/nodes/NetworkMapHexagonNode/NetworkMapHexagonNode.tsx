/*
 * Copyright 2023 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

import React from 'react'
import { Handle, Position, NodeProps } from 'reactflow'
import { Container, Text } from '@harness/uicore'
import type { NetworkMapNodeData } from '@discovery/components/NetworkGraph/types'
import css from './NetworkMapHexagonNode.module.scss'

export default function NetworkMapHexagonNode(props: NodeProps<NetworkMapNodeData>): React.ReactElement {
  return (
    <div className={css.nodeContainer}>
      <div className={css.hexagon}>
        <Container height="inherit" flex={{ align: 'center-center' }}>
          <Text font={{ size: 'xsmall' }} lineClamp={3}>
            {props.data.name}
          </Text>
        </Container>
      </div>

      <Handle isConnectable={props.isConnectable} className={css.targetHandle} type="target" position={Position.Top} />
      <Handle
        isConnectable={props.isConnectable}
        className={css.sourceHandle}
        type="source"
        position={Position.Bottom}
      />
    </div>
  )
}
