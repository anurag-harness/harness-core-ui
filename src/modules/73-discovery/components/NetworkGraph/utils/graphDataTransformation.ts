/*
 * Copyright 2023 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

import type { Node, Edge } from 'reactflow'
import { NodeTypes, nodeGroupOptions, nodeOptions } from '@discovery/components/NetworkGraph/constants'
import type {
  ApiCreateNetworkMapRequest,
  ApiListDiscoveredServiceConnection,
  ApiListDiscoveredService
} from 'services/servicediscovery'
import type {
  NetworkMapEdgeData,
  NetworkMapNodeData,
  ServiceEdgeData,
  ServiceNodeData
} from '@discovery/components/NetworkGraph/types'

export function getGraphNodesFromServiceList(
  serviceList: ApiListDiscoveredService | null
): Node<ServiceNodeData, NodeTypes>[] {
  if (!serviceList) return []

  const namespaces = new Set<string>()
  const graphNodes: Node<ServiceNodeData, NodeTypes>[] = []
  serviceList.items?.map(service => {
    const namespace = service.spec.kubernetes?.namespace
    if (namespace && !namespaces.has(namespace)) {
      namespaces.add(namespace)
    }

    if (service.id)
      graphNodes.push({
        ...nodeOptions,
        id: service.id,
        data: { ...service, name: service.spec.kubernetes?.name ?? 'service' },
        parentNode: namespace
      })
  })

  namespaces.forEach(value => {
    graphNodes.push({
      ...nodeGroupOptions,
      id: value,
      data: { id: value, name: value }
    })
  })

  return graphNodes
}

export function getGraphEdgesFromServiceConnections(
  connectionList: ApiListDiscoveredServiceConnection | null
): Edge<ServiceEdgeData>[] {
  if (!connectionList) return []

  const graphEdges: Edge<ServiceEdgeData>[] = []

  connectionList.items?.map(connection => {
    if (connection.id && connection.sourceID && connection.destinationID)
      graphEdges.push({
        id: connection.id,
        source: connection.sourceID,
        target: connection.destinationID,
        data: { parentNode: connection.sourceNamespace, ...connection }
      })
  })

  return graphEdges
}

export function getGraphNodesFromNetworkMap(
  networkMap: ApiCreateNetworkMapRequest | undefined
): Node<NetworkMapNodeData, NodeTypes>[] {
  if (!networkMap) return []

  const namespaces = new Set<string>()
  const graphNodes: Node<NetworkMapNodeData, NodeTypes>[] = []

  networkMap.resources?.map(service => {
    const namespace = service.kubernetes?.namespace
    if (namespace && !namespaces.has(namespace)) {
      namespaces.add(namespace)
    }

    if (service.id)
      graphNodes.push({
        ...nodeOptions,
        type: NodeTypes.NetworkMapHexagon,
        id: service.id,
        data: service,
        parentNode: namespace
      })
  })

  namespaces.forEach(value => {
    graphNodes.push({
      ...nodeGroupOptions,
      id: value,
      data: { id: value, name: value, kind: 'discoveredservice' }
    })
  })

  return graphNodes
}

export function getGraphEdgesFromNetworkMap(
  networkMap: ApiCreateNetworkMapRequest | undefined
): Edge<NetworkMapEdgeData>[] {
  if (!networkMap) return []

  const graphEdges: Edge<NetworkMapEdgeData>[] = []

  networkMap.connections?.map(connection => {
    if (connection.from?.id && connection.to?.id)
      graphEdges.push({
        id: `${connection.from.id}-${connection.to.id}`,
        source: connection.from.id,
        target: connection.to.id,
        data: { ...connection, parentNode: connection.from.kubernetes?.namespace }
      })
  })

  return graphEdges
}
