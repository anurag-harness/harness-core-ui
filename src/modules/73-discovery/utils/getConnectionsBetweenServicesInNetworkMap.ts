/*
 * Copyright 2023 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

import {
  ApiListDiscoveredServiceConnection,
  DatabaseConnection,
  DatabaseConnectionType,
  DatabaseNetworkMapEntity
} from 'services/servicediscovery'

export default function getConnectionsBetweenServicesInNetworkMap(
  services: DatabaseNetworkMapEntity[] | null | undefined,
  connectionList: ApiListDiscoveredServiceConnection | null
): DatabaseConnection[] | undefined {
  // Checks for new connections between the selected services
  const serviceHashMap = new Map<string, DatabaseNetworkMapEntity>(services?.map(s => [s.id ?? '', s]))

  const connectionBetweenSelectedServices: DatabaseConnection[] | undefined = connectionList?.items
    ?.filter(conn => serviceHashMap.has(conn.sourceID ?? '') && serviceHashMap.has(conn.destinationID ?? ''))
    .map(conn => ({
      from: {
        id: conn.sourceID,
        kind: serviceHashMap.get(conn.sourceID ?? '')?.kind ?? 'discoveredservice',
        name: conn.sourceName,
        kubernetes: {
          namespace: conn.sourceNamespace
        }
      },
      port: conn.destinationPort,
      to: {
        id: conn.destinationID,
        kind: serviceHashMap.get(conn.destinationID ?? '')?.kind ?? 'discoveredservice',
        name: conn.destinationName,
        kubernetes: {
          namespace: conn.destinationNamespace
        }
      },
      type: conn.type as DatabaseConnectionType,
      manual: false
    }))

  return connectionBetweenSelectedServices
}
