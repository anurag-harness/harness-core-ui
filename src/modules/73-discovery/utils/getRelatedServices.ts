/*
 * Copyright 2023 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

import {
  ApiListDiscoveredServiceConnection,
  ApiListDiscoveredService,
  DatabaseDiscoveredServiceCollection
} from 'services/servicediscovery'

export function getRelatedServices(
  serviceID: string,
  serviceList: ApiListDiscoveredService | null,
  connectionList: ApiListDiscoveredServiceConnection | null
): DatabaseDiscoveredServiceCollection[] | undefined {
  if (!serviceList || !connectionList) return

  const relatedServiceIds = connectionList.items?.map(conn => {
    if (serviceID === conn.sourceID) return conn.destinationID ?? ''
    /* istanbul ignore next */
    if (serviceID === conn.destinationID) return conn.sourceID ?? ''
  })

  return serviceList?.items?.filter(service => service.id === serviceID || relatedServiceIds?.includes(service.id))
}
