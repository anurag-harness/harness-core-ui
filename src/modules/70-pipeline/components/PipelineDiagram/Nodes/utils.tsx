/*
 * Copyright 2022 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

import { isNumber } from 'lodash-es'
import { yamlStringify } from '@common/utils/YamlHelperMethods'
import { dragPlaceholderImageBase64 } from './assets/dragImageBase64'
import { dragStagePlaceholderImageBase64 } from './assets/dragStageImageBase64'
import defaultCss from './DefaultNode/DefaultNode.module.scss'

interface DotNotationPathProps {
  identifier: string
  entityType?: string
  baseDotNotation?: string
}

export enum NodeEntity {
  STAGE = 'STAGE',
  STEP = 'STEP'
}
export enum NodeWrapperEntity {
  step = 'step',
  stepGroup = 'stepGroup',
  stage = 'stage'
}

export const COLLAPSED_MATRIX_NODE_LENGTH = 8
export const MAX_ALLOWED_MATRIX_COLLAPSED_NODES = 4
export const DEFAULT_MATRIX_PARALLELISM = 1

export const getPositionOfAddIcon = (props: any, isRightNode?: boolean): string => {
  if (isRightNode) {
    return '-40px'
  }
  if (props?.children?.length) {
    if (props?.prevNode?.children?.length) {
      return '-65px'
    }
    if (props?.prevNode) return '-58px'
  }
  if (props?.prevNode?.children?.length) {
    return '-58px'
  }
  if (props?.parentIdentifier && !props.prevNode) {
    return '-35px'
  }
  return '-50px'
}

export const transformMatrixLabels = (nodeData: string): string => {
  const parsedNodeName = {}
  try {
    // if object parse to yaml
    const parsedVal = JSON.parse(nodeData)
    if (parsedVal && !isNumber(parsedVal)) {
      Object.assign(parsedNodeName, parsedVal)
      return yamlStringify(parsedNodeName, { indent: 4 })
    } else {
      return JSON.stringify(nodeData)
    }
  } catch (_e) {
    // name is string/number, parse it to string
    return JSON.stringify(nodeData)
  }
}

export const attachDragImageToEventHandler = (event: React.DragEvent<HTMLDivElement>, type?: NodeEntity): void => {
  // set drag image preview to custom icon in case of safari browser, as safari blocks image preview if dom tree have transform property on any parent
  if (navigator.userAgent.search('Safari') >= 0 && navigator.userAgent.search('Chrome') < 0) {
    const dragImage = document.createElement('img')
    dragImage.src = type === NodeEntity.STAGE ? dragStagePlaceholderImageBase64 : dragPlaceholderImageBase64
    dragImage.style.position = 'absolute'
    dragImage.style.left = '-100%'

    document.body.append(dragImage)
    setTimeout(() => dragImage.remove(), 0)

    event.dataTransfer.setDragImage(dragImage, dragImage.width / 2, dragImage.height / 2)
  }
}

export const getEntityIdentifierBasedDotNotationPath = ({
  baseDotNotation,
  identifier,
  entityType = NodeWrapperEntity.step
}: DotNotationPathProps): string => {
  return `${baseDotNotation}.${entityType}.${identifier}`
}

export const getBaseDotNotationWithoutEntityIdentifier = (dotNotation = ''): string => {
  const lastDotIndex = dotNotation.lastIndexOf('.')
  return lastDotIndex !== -1 ? dotNotation.substring(0, lastDotIndex) : dotNotation
}

export const getConditionalClassName = (isDisabled: boolean, className?: string): { className?: string } => {
  return isDisabled ? { className: defaultCss.disabledIcon } : className ? { className } : {}
}
