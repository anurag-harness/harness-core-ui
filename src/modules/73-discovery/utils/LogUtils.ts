/*
 * Copyright 2023 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

import { sanitizeHTML } from '@common/utils/StringUtils'
import { formatDatetoLocale } from '@common/utils/dateUtils'
import { loggerFor } from 'framework/logging/logging'
import { ModuleName } from 'framework/types/ModuleName'
import { DatabaseInstallationCollection } from 'services/servicediscovery'

export interface ArgsType {
  PodName: string
  PodNamespace: string
}

export interface DAgentLogs {
  clusterLogs: string
  nodeLogs: string
}

/**
 * Data for a log unit/section
 */
export interface LogSectionData {
  /**
   * The log data
   */
  data: LogLineData[]
}

/**
 * Keys in the json from server
 */
export type TextKeys = 'level' | 'out' | 'time' | 'args'

export interface LogLineData {
  /**
   * Text for the line
   */
  text: Partial<Record<TextKeys, string | ArgsType>>
  /**
   * Indices of the search result with in the current line.
   * This will be a continous list
   *
   * If this values is `[3, 4]`, it means the result number 3 and 4 will be found in this line
   */
  searchIndices?: Partial<Record<TextKeys, number[]>>
}

const logger = loggerFor(ModuleName.CHAOS)

export function processLogsData(data: string): LogLineData[] {
  return String(data)
    .split('\n')
    .reduce<LogSectionData['data']>((accumulator, line) => {
      /* istanbul ignore else */
      if (line.length > 0) {
        try {
          const { level, time, out, args } = JSON.parse(line) as Record<string, string>
          accumulator.push({
            text: {
              level: sanitizeHTML(level),
              time: formatDatetoLocale(time),
              out: sanitizeHTML(out),
              args: args
            }
          })
        } catch (error) {
          logger.error(`Error while parsing logs: ${error}`)
        }
      }
      return accumulator
    }, [])
}

export function processAgentLogs(
  logData: string | undefined,
  installationList: DatabaseInstallationCollection | undefined
): DAgentLogs {
  const logsData = processLogsData(logData ?? '')
  let clusterLogs = ''
  let nodeLogs = ''
  const nodeNames: Array<string> = []
  installationList?.agentDetails?.node?.forEach(node => {
    if (node?.name) nodeNames.push(node?.name)
  })
  logsData.map(log => {
    if (installationList?.agentDetails?.cluster?.name === (log.text.args as ArgsType)?.PodName) {
      clusterLogs = clusterLogs + log.text.out + '\n'
    }
    if (nodeNames.includes((log.text.args as ArgsType)?.PodName)) {
      nodeLogs = nodeLogs + log.text.out + '\n'
    }
  })

  return {
    clusterLogs: clusterLogs.length ? clusterLogs : 'No cluster level logs found',
    nodeLogs: nodeLogs.length ? nodeLogs : 'No node level logs found'
  }
}
