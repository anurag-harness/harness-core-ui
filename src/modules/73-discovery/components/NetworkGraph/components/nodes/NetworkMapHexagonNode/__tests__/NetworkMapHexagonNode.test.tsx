/*
 * Copyright 2023 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

import React from 'react'
import { render } from '@testing-library/react'
import { NodeProps, ReactFlowProvider } from 'reactflow'
import { TestWrapper } from '@common/utils/testUtils'
import NetworkMapHexagonNode from '../NetworkMapHexagonNode'

const props: NodeProps = {
  id: 'a1',
  data: { id: 'a1', name: 'a1', kind: 'discoveredservice' },
  isConnectable: true,
  dragging: false,
  selected: false,
  type: 'hexagon',
  xPos: 0,
  yPos: 0,
  zIndex: 1
}

describe('NetworkMapHexagonNode', () => {
  test('render component with mock data', async () => {
    const { container } = render(
      <TestWrapper>
        <ReactFlowProvider>
          <NetworkMapHexagonNode {...props} />
        </ReactFlowProvider>
      </TestWrapper>
    )
    expect(container).toMatchSnapshot()
  })

  test('render component with mock data with data field missing name', async () => {
    const newProps = { ...props, data: { ...props.data, name: '' } }
    const { container } = render(
      <TestWrapper>
        <ReactFlowProvider>
          <NetworkMapHexagonNode {...newProps} />
        </ReactFlowProvider>
      </TestWrapper>
    )
    expect(container).toMatchSnapshot()
  })
})
