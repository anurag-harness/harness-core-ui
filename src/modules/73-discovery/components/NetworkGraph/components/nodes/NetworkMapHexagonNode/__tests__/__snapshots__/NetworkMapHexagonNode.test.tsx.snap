// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NetworkMapHexagonNode render component with mock data 1`] = `
<div>
  <div
    class="nodeContainer"
  >
    <div
      class="hexagon"
    >
      <div
        class="StyledProps--font StyledProps--main StyledProps--flex StyledProps--flex-align-center-center"
        style="height: inherit;"
      >
        <p
          class="StyledProps--font Text--lineclamp StyledProps--main StyledProps--font-size-xsmall"
          style="--text-line-clamp: 3;"
        >
          a1
        </p>
      </div>
    </div>
    <div
      class="react-flow__handle react-flow__handle-top nodrag nopan targetHandle target connectable connectablestart connectableend connectionindicator"
      data-handlepos="top"
      data-id="null-null-target"
    />
    <div
      class="react-flow__handle react-flow__handle-bottom nodrag nopan sourceHandle source connectable connectablestart connectableend connectionindicator"
      data-handlepos="bottom"
      data-id="null-null-source"
    />
  </div>
</div>
`;

exports[`NetworkMapHexagonNode render component with mock data with data field missing name 1`] = `
<div>
  <div
    class="nodeContainer"
  >
    <div
      class="hexagon"
    >
      <div
        class="StyledProps--font StyledProps--main StyledProps--flex StyledProps--flex-align-center-center"
        style="height: inherit;"
      >
        <p
          class="StyledProps--font Text--lineclamp StyledProps--main StyledProps--font-size-xsmall"
          style="--text-line-clamp: 3;"
        >
          
        </p>
      </div>
    </div>
    <div
      class="react-flow__handle react-flow__handle-top nodrag nopan targetHandle target connectable connectablestart connectableend connectionindicator"
      data-handlepos="top"
      data-id="null-null-target"
    />
    <div
      class="react-flow__handle react-flow__handle-bottom nodrag nopan sourceHandle source connectable connectablestart connectableend connectionindicator"
      data-handlepos="bottom"
      data-id="null-null-source"
    />
  </div>
</div>
`;
